be/src/runtime/exec_env.cpp

```diff
@@ -624,4 +624,8 @@ int32_t ExecEnv::calc_pipeline_dop(int32_t pipeline_dop) const {
     return std::max<int32_t>(1, _max_executor_threads / 2);
 }
 
+ThreadPool* ExecEnv::vacuum_thread_pool() {
```

Comment:
I see some callers don't check if the return is nullptr.
How about returning an invalid thread pool rather than a nullptr to avoid a nullptr access exception?

Reply:
Hi @imay, for simplicity, I added NULL checks for the vacuum thread pool, PTAL.